src/release_artifacts_resources/ios/cdk/cdk/credential_rotation/lambda_functions/src/source_data_generator/aws_session_credential_source.py

```diff
@@ -0,0 +1,92 @@
+"""Creates static credentials, uses them to adopt the specified role and get temporary
+session credentials, and deletes the static credentials.
+"""
+
+import os
+from datetime import datetime
+from time import sleep
+from typing import Dict, Tuple
+
+import boto3
+from src.utils.retry import retry
+
+DEFAULT_REGION = "us-west-2"
+
+# Make sure this is at least 2 hours longer than the key rotation schedule set up in the CFn.
+# Currently, "2 hours" is an arbitrary value based on the fact that a few of our deployments
+# take ~2 hour to run, so we want to leave sufficient time for executing the tests in cases where a
+# test is invoked with session credentials created with access keys that are rotated immediately
+# thereafter.
+SESSION_DURATION_SECONDS = 14400  # 4 hours
+
+# Provided by Lambda
+REGION = os.environ.get("AWS_REGION", DEFAULT_REGION)
+
+
+def generate_session_credentials(configuration: map, iam=None, sts=None) -> Dict[str, str]:
+
+    if not configuration:
+        raise RuntimeError("Configuration is required to generate session credentials")
+
+    iam_user_key = configuration["user_env_variable"]
+    iam_role_key = configuration["iam_role_env_variable"]
+    iam_username = os.environ.get(iam_user_key)
+    iam_role = os.environ.get(iam_role_key)
+
+    user_credentials: Tuple[str, str] = ()
+    session_credentials: Dict[str, str] = {}
+
+    try:
+
+        iam_client = iam or boto3.client("iam", region_name=REGION)
+        user_credentials = create_user_credentials(iam_username, iam=iam_client)
+        wait_for_user_credentials()
+        session_credentials = get_session_credentials(user_credentials, role_arn=iam_role, sts=sts)
+
+    finally:
+        if user_credentials:
+            iam_client.delete_access_key(UserName=iam_username, AccessKeyId=user_credentials[0])
+    return session_credentials
```

Comment:
nit, add vertical whitespace
```suggestion

    return session_credentials
```